docs/reference/modules/cluster/remote-clusters-migration.asciidoc

```diff
@@ -0,0 +1,242 @@
+[[remote-clusters-migrate]]
+=== Migrate remote clusters from certificate to API key authentication
+
+++++
+<titleabbrev>Migrate from certificate to API key authentication</titleabbrev>
+++++
+
+The API key based security model for remote clusters offers administrators more
+fine-grained access controls compared to the TLS certificate based security
+model. For that reason, you may want to migrate from the certificate based
+security model to the API key based model.
+
+While it is possible to migrate by defining a new remote cluster connection,
+using a new alias, this has several downsides:
+
+- It's not possible to change the leader cluster alias for existing {ccr} tasks.
+As a result, with a new remote cluster, follower indices would need to be
+re-created from scratch.
+- Transform and anomaly detection jobs do allow updating the remote cluster
+alias. However, if the job was created with wildcards, for example 
+`*:source_index`, and `superuser`, adding a new remote cluster will cause the 
+job to do double the amount of work and potentially skew results with 
+duplications.
+
+For these reasons, you may prefer to migrate a remote cluster in-place, by
+following these steps:
+
+. <<remote-clusters-migration-prerequisites,Review the prerequisites>>
+. <<remote-clusters-migration-remote-cluster>>
+. <<remote-clusters-migration-stop>>
+. <<remote-clusters-migration-reconnect>>
+. <<remote-clusters-migration-resume>>
+. <<remote-clusters-migration-disable-cert>>
+
+[[remote-clusters-migration-prerequisites]]
+==== Prerequisites
+
+* The nodes of the local and remote clusters must be on version 8.10 or later.
+* The local and remote clusters must have an appropriate license. For more
+information, refer to https://www.elastic.co/subscriptions.
+
+[[remote-clusters-migration-remote-cluster]]
+==== Reconfigure the remote cluster and generate a cross-cluster API key
+
+On the remote cluster:
+
+include::remote-clusters-api-key.asciidoc[tag=remote-cluster-steps]
+
+[[remote-clusters-migration-stop]]
+==== Stop cross-cluster operations
+
+On the local cluster, stop any persistent tasks that refer to the remote 
+cluster:
+
+* Use the <<stop-transform>> API to stop any transforms.
+* Use the <<ml-close-job>> API to close any anomaly detection jobs.
+* Use the <<ccr-pause-auto-follow-pattern>> API to pause any auto-follow {ccr}.
+* Use the <<ccr-post-pause-follow>> API to pause any manual {ccr} or existing
+indices that were created from the auto-follow pattern.
+
+[[remote-clusters-migration-reconnect]]
+==== Reconnect to the remote cluster
+
+On the local cluster:
+
+. Enhance any roles used by local cluster users with the required
+<<roles-remote-indices-priv,remote indices privileges>> for {ccr} and {ccs}.
+Refer to <<remote-clusters-privileges-api-key>>. Note:
+** You only need to assign additional `remote_indices` privileges to existing
+roles.
```

Comment:
For RCS 1.0, role definitions are resolved on the remote cluster. This means, the role definitions may not even exist in the local cluster. Or if they exist, they might be different. [Our docs](https://elasticsearch_98999.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/remote-clusters-cert.html#remote-clusters-privileges-cert) in fact suggest this setup. So the new `remote_indices` privileges should match the `indices` privileges from the same role but located on the remote cluster. I hope we could make this point clear.

Reply:
In https://github.com/elastic/elasticsearch/pull/98999/commits/ea4584c30623f194ca88a1098fd3b27886030e68, I've changed the bullet into:

_** You only need to assign additional `remote_indices` privileges to existing
roles used for cross-cluster operations. You should be able to copy these
privileges from the original roles on the remote cluster, where they are defined
under the certification based security model._

I also thought it should be good to emphasize that these roles can't exceed the `access` privilege granted by
the cross-cluster API key, so I've also added the following bullet:

_** The roles on the local cluster can't exceed the `access` privilege granted by
the cross-cluster API key._